tests/models/udop/test_tokenization_udop.py

```diff
@@ -1888,3 +1888,13 @@ def test_text_target(self):
         self.assertListEqual(encoding_p["attention_mask"], [1, 1, 1])
         self.assertDictEqual(dict(encoding_p), dict(encoding_r))
         self.assertEqual(tokenizer_p.decode(encoding_p["input_ids"]), expected_decoding)
+
+    def test_special_tokens(self):
+        tokenizer_r = UdopTokenizerFast.from_pretrained("microsoft/udop-large")
+
+        ids = [0, 8986, 32942, 32966, 32554, 32551, 1]
+        decoding = tokenizer_r.decode(ids)
+
+        excepted_decoding = "<pad> paragraph<loc_58><loc_34><loc_446><loc_449></s>"
```

Comment:
@ArthurZucker I've noticed that the slow Python tokenizer returns this:

```
<pad>paragraph <loc_58> <loc_34> <loc_446> <loc_449> </s>
```
in other words, with spaces in between the special tokens. Does this have anything to do with rstrip/lstrip?

Reply:
`spaces_between_special_tokens=True` is the defaults 😢

Reply:
make sure to test both!

Reply:
We should test `tokenizer_p.decode(ids, spaces_between_special_tokens = False)`. If that is not the issue it's decoding function that is at fault!

Reply:
Ok I've tested this, but it results in different decodings:
```
assert decoding_p == decoding_r == excepted_decoding
        AssertionError: assert '<pad>paragra...<loc_449></s>' == '<pad> paragr...<loc_449></s>'
          - <pad> paragraph<loc_58><loc_34><loc_446><loc_449></s>
          ?      -
         + <pad>paragraph<loc_58><loc_34><loc_446><loc_449></s>
```
Looks like the slow tokenizer adds a space after the /<pad> token